clayui.com/src/components/Hooks/useStickyState.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import React from 'react';
+
+const inBrowser = typeof window !== 'undefined';
+
+const localStorage = {
+	getItem(key) {
+		return inBrowser ? window.localStorage.getItem(key) : null;
+	},
+
+	setItem(key, value) {
+		if (inBrowser) {
+			window.localStorage.setItem(key, value);
+		}
+	},
+};
+
+function useStickyState(defaultValue, key) {
+	const [value, setValue] = React.useState(() => {
+		let stickyValue;
+
+		try {
+			stickyValue = localStorage.getItem(key);
+		} catch (error) {
+			stickyValue = null;
+		}
+
+		return stickyValue !== null ? JSON.parse(stickyValue) : defaultValue;
```

Comment:
One thing to bear in mind is that `JSON.parse` can throw too (eg. someone sticks garbage in their local storage → BOOM), so you might move that into the `try` too:

```js
const [value, setValue] = React.useState(() => {
    try {
        const value = localStorage.getItem(key);

        return value === null ? defaultValue : JSON.parse(value);
    } catch (error) {
        return defaultValue;
    }
});
```

You can omit the `(error)` entirely if we have Babel set up to allow optional `catch` binding in this repo (but I don't know if we do).